### ISSUE TITLE
Runner MODULE_NOT_FOUND

### ISSUE THREAD

**Issue by nichlaes** [INTO-CPS-Association/DTaaS#436]
When running @into-cps-association/runner@0.0.1 in docker container, I get the following error message:

```

2024-01-10 11:29:41 node:internal/modules/esm/resolve:255
2024-01-10 11:29:41     throw new ERR_MODULE_NOT_FOUND(
2024-01-10 11:29:41           ^
2024-01-10 11:29:41 
2024-01-10 11:29:41 Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/usr/local/lib/node_modules/@into-cps-association/runner/dist/src/app.controller' imported from /usr/local/lib/node_modules/@into-cps-association/runner/dist/src/app.module.js
2024-01-10 11:29:41     at finalizeResolution (node:internal/modules/esm/resolve:255:11)
2024-01-10 11:29:41     at moduleResolve (node:internal/modules/esm/resolve:908:10)
2024-01-10 11:29:41     at defaultResolve (node:internal/modules/esm/resolve:1121:11)
2024-01-10 11:29:41     at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12)
2024-01-10 11:29:41     at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25)
2024-01-10 11:29:41     at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38)
2024-01-10 11:29:41     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39)
2024-01-10 11:29:41     at link (node:internal/modules/esm/module_job:84:36) {
2024-01-10 11:29:41   code: 'ERR_MODULE_NOT_FOUND',
2024-01-10 11:29:41   url: 'file:///usr/local/lib/node_modules/@into-cps-association/runner/dist/src/app.controller'
2024-01-10 11:29:41 }
2024-01-10 11:29:41 
2024-01-10 11:29:41 Node.js v20.10.0
```

**Comment by nichlaes** [comment]
dockerfile

```
FROM node:20.10.0-slim

#! docker should be run from the root directory of the project

# Set the working directory inside the container
WORKDIR /dtaas/runner

# pull the libms package
RUN npm i -g @into-cps-association/runner@0.0.1

# Define the command to run your app
CMD ["runner"]

```

**Comment by prasadtalasila** [comment]
The problem can be seen in the DTaaS user workspace as well. From within the userworkspace, install nodejs v20.0 and then

```bash
$node install -g @into-cps-association/runner
runner -h
node:internal/modules/esm/resolve:264
    throw new ERR_MODULE_NOT_FOUND(
          ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/usr/lib/node_modules/@into-cps-association/runner/dist/src/app.controller' imported from /usr/lib/node_modules/@into-cps-association/runner/dist/src/app.module.js
    at finalizeResolution (node:internal/modules/esm/resolve:264:11)
    at moduleResolve (node:internal/modules/esm/resolve:917:10)
    at defaultResolve (node:internal/modules/esm/resolve:1130:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39)
    at link (node:internal/modules/esm/module_job:84:36) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///usr/lib/node_modules/@into-cps-association/runner/dist/src/app.controller'
}

Node.js v20.11.0
(base)

```

**Comment by prasadtalasila** [comment]
This error is due to upgrade of NodeJS from v18.x to v20.x. The cause is yet unknown.

**Comment by prasadtalasila** [comment]
problem resolved in commit 375d3a9bc34231d41761e32d3d395ca0985d9311.
